server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -2368,54 +2376,69 @@ protected ServiceOfferingVO createServiceOffering(final long userId, final boole
                 limitResourceUse, volatileVm, displayText, typedProvisioningType, localStorageRequired, false, tags, isSystem, vmType,
                 domainId, hostTag, deploymentPlanner);
 
-        if (isCustomizedIops != null) {
-            bytesReadRate = null;
-            bytesWriteRate = null;
-            iopsReadRate = null;
-            iopsWriteRate = null;
-
-            if (isCustomizedIops) {
+        if (Boolean.TRUE.equals(isCustomizedIops)) {
```

Comment:
In this section, the previous version of the code treated isCustomizedIops null value distinctly from isCustomizedIops false value.  I am reasonably sure they should be treated the same, and also it nulled out values for bytesReadRate / bytesWriteRate .  I'd love some other eyes on this, as I'm not entirely sure what the original intent of the code was and I may be stepping on someone's expected behavior here

Reply:
@mike-tutkowski can you comment?